contrib/net/http/trace.go

```diff
@@ -64,12 +64,18 @@ func TraceAndServe(h http.Handler, w http.ResponseWriter, r *http.Request, cfg *
 		opts = append(opts, tracer.ChildOf(spanctx))
 	}
 	span, ctx := tracer.StartSpanFromContext(r.Context(), "http.request", opts...)
-	defer span.Finish(cfg.FinishOpts...)
+	rw, ddrw := wrapResponseWriter(w, span)
+	defer func() {
+		if ddrw.status == 0 {
+			span.SetTag(ext.HTTPCode, "200")
+		}
```

Comment:
WDYT of moving the entire status code monitoring here by moving https://github.com/DataDog/dd-trace-go/blob/v1/contrib/net/http/trace.go#L110-L113 too.

That would allow to perform the same optimization as https://github.com/DataDog/dd-trace-go/pull/1183/files in order to avoid avoid the stack trace in this case that I think is useless since it happens within our middleware and out of the user handler.